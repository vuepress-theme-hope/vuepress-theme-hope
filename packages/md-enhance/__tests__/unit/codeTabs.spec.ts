import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { codeTabs } from "../../src/node/markdown-it/index.js";

const markdownIt = MarkdownIt({ linkify: true }).use(codeTabs);

describe("code tabs", () => {
  it("Should render single block", () => {
    expect(
      markdownIt.render(`
::: code-tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("Should render multiple block", () => {
    expect(
      markdownIt.render(`
::: code-tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

@tab ts

\`\`\`ts
const a = 1;
\`\`\`

:::
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
@tab ts
\`\`\`ts
const a = 1;
\`\`\`
:::
`)
    ).toMatchSnapshot();
  });

  it("Should support id", () => {
    expect(
      markdownIt.render(`
::: code-tabs#event

@tab js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs#event-id
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs#id with space
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs # id starts and having space in the end
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("Should support active", () => {
    expect(
      markdownIt.render(`
::: code-tabs

@tab:active js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab:active js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

@tab:active ts

\`\`\`ts
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
@tab:active ts
\`\`\`ts
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("Should support value", () => {
    expect(
      markdownIt.render(`
::: code-tabs

@tab js#javascript

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab:active js#javascript
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs

@tab js#js

\`\`\`js
const a = 1;
\`\`\`

@tab:active ts #typescript

\`\`\`ts
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab js # javascript
\`\`\`js
const a = 1;
\`\`\`
@tab:active ts #typescript
\`\`\`ts
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("should ignore other items", () => {
    expect(
      markdownIt.render(`
::: code-tabs

\`\`\`coffee
const a = 1;
\`\`\`

@tab:active js

\`\`\`js
const a = 1;
\`\`\`

\`\`\`ts
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
\`\`\`coffee
const a = 1;
\`\`\`
@tab:active js
\`\`\`js
const a = 1;
\`\`\`
\`\`\`ts
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs

@tab js

A text

\`\`\`js
const a = 1;
\`\`\`

Another text

@tab:active ts

Another text again

\`\`\`ts
const a = 1;
\`\`\`

Another text again

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: code-tabs
@tab js
A text
\`\`\`js
const a = 1;
\`\`\`
Another text
@tab:active ts
Another text again
\`\`\`ts
const a = 1;
\`\`\`
Another text again
:::
    `)
    ).toMatchSnapshot();
  });
});
