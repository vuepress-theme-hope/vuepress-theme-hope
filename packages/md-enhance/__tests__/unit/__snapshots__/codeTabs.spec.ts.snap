// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`code tabs > Should render multiple block 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should render multiple block 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should render single block 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should render single block 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support active 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support active 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support active 3`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support active 4`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support id 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"event\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support id 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"event-id\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support id 3`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"id with space\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support id 4`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"id starts and having space in the end\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support value 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\",\\"id\\":\\"javascript\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support value 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\",\\"id\\":\\"javascript\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support value 3`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\",\\"id\\":\\"js\\"},{\\"title\\":\\"ts\\",\\"id\\":\\"typescript\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > Should support value 4`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\",\\"id\\":\\"javascript\\"},{\\"title\\":\\"ts\\",\\"id\\":\\"typescript\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > should ignore other items 1`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > should ignore other items 2`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > should ignore other items 3`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;

exports[`code tabs > should ignore other items 4`] = `
"<CodeTabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</CodeTabs>
"
`;
